File: 08-DesignPatterns-CSharp/08-1-1-Observer-Solution/DebugView.cs
namespace Observer
{
    using System.Diagnostics;

    public sealed class DebugView : IView
    {
        private readonly int _id;

        public DebugView(int id)
        {
            _id = id;
        }

        public void Update(float temperature, float humidity, float windSpeed, float rainProbability)
        {
            Debug.WriteLine($"View #{_id} | Temperature: {temperature} C° | Humidity: {humidity} % | Windspeed: {windSpeed} m/s | Rain Probability: {rainProbability} %");
        }
    }
}

File: 08-DesignPatterns-Java/Solution/Observer/src/DebugView.java
public class DebugView implements IView {
    private final int _id;

    public DebugView(int id)
    {
        _id = id;
    }

    @Override
    public void update(float temperature, float humidity, float windSpeed, float rainProbability) {
        // AFAIK there is no Debug-PrintStream in Java. But I am not an expert in Java... :-)
        System.out.println("DEBUG | View #" + _id + " | Temperature: " + temperature + " C° | Humidity: " + humidity + " % | Windspeed: " + windSpeed + " m/s | Rain Probability: " + rainProbability + "%");
    }
}
